File: Data_Structure/Python/BubbleSort.py
def Bubblesort(arr):
    size = len(arr)
    for step in range(size - 1):
        for i in range(size - step - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]

def printArray(arr):
    for num in arr:
        print(num, end=" ")
    print()

print("Enter the number of elements in the list:")
n = int(input())
arr = []
print("Enter the elements:")
for i in range(n):
    arr.append(int(input()))

print("Unsorted Array:")
printArray(arr)

Bubblesort(arr)
print("Sorted Array:")
printArray(arr)

File: Data_Structure/Java/BubbleSort.java
import java.util.Scanner;

public class BubbleSort {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter number of elements:");
        int n = sc.nextInt();

        int[] arr = new int[n];
        System.out.println("Enter " + n + " elements:");
        for(int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }

        System.out.println("Unsorted Array:");
        printArray(arr);

        sort(arr);
        System.out.println("Sorted Array:");
        printArray(arr);

        sc.close();
    }

    public static void sort(int[] arr){
        int size = arr.length;
        for(int step = 0; step < size - 1; step++){
            for(int i = 0; i < size-step-1; i++){
                if(arr[i] > arr[i+1]){
                    int temp = arr[i];
                    arr[i] = arr[i+1];
                    arr[i+1] = temp;
                }
            }
        }
    }

    public static void printArray(int[] arr){
        for(int num : arr){
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
